lib/spack/spack/test/cmd/analyze.py

```diff
@@ -0,0 +1,125 @@
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+
+
+import spack.config
+import spack.package
+import spack.cmd.install
+
+import spack.util.spack_json as sjson
+from spack.main import SpackCommand
+
+install = SpackCommand('install')
+analyze = SpackCommand('analyze')
+
+
+def test_test_package_not_installed(mock_fetch, install_mockery_mutable_config):
+
+    # We cannot run an analysis for a package not installed
+    out = analyze('run', 'libdwarf', fail_on_error=False)
+    assert "==> Error: Spec 'libdwarf' matches no installed packages.\n" in out
+
+
+def test_analyze_output(tmpdir, mock_fetch, install_mockery_mutable_config):
+    """Test that an analyzer is able to run and errors if a run is requested
+    for one that does not exist
+    """
+    install('libdwarf')
+    analyzer_dir = tmpdir.join('analyzers')
+
+    # An analyzer that doesn't exist should not work
+    out = analyze('run', '-a', 'pusheen', 'libdwarf', fail_on_error=False)
+    assert '==> Error: Analyzer pusheen does not exist\n' in out
+
+    # We will output to this analyzer directory
+    analyzer_dir = tmpdir.join('analyzers')
+    out = analyze('run', '-a', 'install_files', '-p', str(analyzer_dir), 'libdwarf')
+
+    # Ensure that if we run again without over write, we don't run
+    out = analyze('run', '-a', 'install_files', '-p', str(analyzer_dir), 'libdwarf')
+    assert "skipping" in out
+
+    # With overwrite it should run
+    out = analyze('run', '-a', 'install_files', '-p', str(analyzer_dir),
+                  '--overwrite', 'libdwarf')
+    assert "==> Writing result to" in out
+
+
+def _run_analyzer(name, package, tmpdir):
+    """A shared function to test that an analyzer runs. We return the output
+    file for further inspection.
+    """
+    analyzer = spack.analyzers.get_analyzer(name)
+    analyzer_dir = tmpdir.join('analyzers')
+    out = analyze('run', '-a', analyzer.name, '-p', str(analyzer_dir), package)
+
+    assert "==> Writing result to" in out
+    assert "/%s/%s\n" % (analyzer.name, analyzer.outfile) in out
+
+    # The output file should exist
+    output_file = out.strip('\n').split(' ')[-1].strip()
+    assert os.path.exists(output_file)
+    return output_file
+
+
+def test_installfiles_analyzer(tmpdir, mock_fetch, install_mockery_mutable_config):
+
+    """test the install files analyzer
+    """
+    install('libdwarf')
+    output_file = _run_analyzer("install_files", "libdwarf", tmpdir)
+
+    # Ensure it's the correct content
+    with open(output_file, 'r') as fd:
+        content = sjson.load(fd.read())
+
+    basenames = set()
+    for key, attrs in content.items():
+        basenames.add(os.path.basename(key))
+
+    # Check for a few expected files
+    for key in ['.spack', 'libdwarf', 'packages', 'repo.yaml', 'repos']:
+        assert key in basenames
+
+
+def test_environment_analyzer(tmpdir, mock_fetch, install_mockery_mutable_config):
+
+    """test the environment variables analyzer
```

Comment:
I think I probably started with two lines of args, like:
```python
def test_install_output_on_build_error(mock_packages, mock_archive, mock_fetch,
                                       config, install_mockery, capfd):
    """This test used to assume receiving full output, but since we've updated
```
and then removed most of the pytest fixures and left the line empty, and then proceeded to copy paste it. Will fix asap!